src/test/java/io/xpring/xrpl/ProtocolBufferConversionTest.java

```diff
@@ -119,4 +116,43 @@ public void convertIssuedCurrencyWithBadValueTest() {
         // THEN the result is null
         assertThat(xrpIssuedCurrency).isNull();
     }
+
+
+    // CurrencyAmount
+
+    @Test
+    public void convertCurrencyAmountWithDropsTest() {
+        // GIVEN a currency amount protocol buffer with an XRP amount.
+        // WHEN the protocol buffer is converted to a native Java type.
+        XRPCurrencyAmount xrpCurrencyAmount = XRPCurrencyAmount.from(FakeXRPProtobufs.dropsCurrencyAmount);
+
+        // Then the result has drops set and an empty issued currency.
+        assertThat(xrpCurrencyAmount.drops())
+                .isEqualTo(Long.toString(FakeXRPProtobufs.dropsCurrencyAmount.getXrpAmount().getDrops()));
+        assertThat(xrpCurrencyAmount.issuedCurrency()).isNull();
+    }
+
+    @Test
+    public void convertCurrencyAmountWithIssuedCurrency() {
+        // GIVEN a currency amount protocol buffer with an issued currency amount.
+        // WHEN the protocol buffer is converted to a native Java type.
+        XRPCurrencyAmount xrpCurrencyAmount = XRPCurrencyAmount.from(FakeXRPProtobufs.issuedCurrencyCurrencyAmount);
+
+        // THEN the result has an issued currency set and no drops amount.
+        assertThat(xrpCurrencyAmount.drops()).isNull();
+        assertThat(xrpCurrencyAmount.issuedCurrency())
+                .isEqualTo(XRPIssuedCurrency.from(FakeXRPProtobufs.issuedCurrencyCurrencyAmount.getIssuedCurrencyAmount()));
+    }
+
+    @Test
+    public void convertCurrencyAmountWithBadInputsTest() {
```

Comment:
There's actually another case here where the issued currency input is invalid (which is what we've been testing in other tests). This test case is valid though and we should probably have it. Mind adding a test case for an invalidIssuedCurrency?

Extra credit if you want to add it to JS / Swift, otherwise we can file an Asana issue and reference this comment and we can address at some point in the future.

Reply:
Great, test case has been added here and can do in JS and Swift as well